### PR TITLE
Fix: Tutorial config-based invalid keywords #20071

### LINKED ISSUE #18663: Low-code CDK: Tutorial Bug
## Environment
- **Airbyte version**:  0.40.17
- **OS Version / Instance**: macOS
- **Deployment**: Docker
- **Step where error happened**: [Tutorial Step 5: Incremental Reads](https://docs.airbyte.com/connector-development/config-based/tutorial/incremental-reads)

## Current Behavior
I copy the full `spec.yaml` and `exchange_rate_tutorial,.yaml` from tutorial step 5 (reproduced fully in Appendix below).

Then I run
```
python main.py read --config secrets/config.json --catalog integration_tests/configured_catalog.json
```
and obtain the errors attached under "Logs".

## Expected Behavior
No error.

## Logs

[log.txt](https://github.com/airbytehq/airbyte/files/9895960/log.txt)

## Steps to Reproduce
Follow the tutorial steps 1 through 5. (Steps 1 to 4 work exactly as intended.)

## Are you willing to submit a PR?
Yes, with guidance!

## Appendix
The full `spec.yaml`:
```
documentationUrl: https://docs.airbyte.io/integrations/sources/exchangeratesapi
connectionSpecification:
  $schema: http://json-schema.org/draft-07/schema#
  title: exchangeratesapi.io Source Spec
  type: object
  required:
    - start_date
    - access_key
    - base
  additionalProperties: true
  properties:
    start_date:
      type: string
      description: Start getting data from that date.
      pattern: ^[0-9]{4}-[0-9]{2}-[0-9]{2}$
      examples:
        - YYYY-MM-DD
    access_key:
      type: string
      description: >-
        Your API Access Key. See <a
        href="https://exchangeratesapi.io/documentation/">here</a>. The key is
        case sensitive.
      airbyte_secret: true
    base:
      type: string
      description: >-
        ISO reference currency. See <a
        href="https://www.ecb.europa.eu/stats/policy_and_exchange_rates/euro_reference_exchange_rates/html/index.en.html">here</a>.
      examples:
        - EUR
        - USD
```

and the full `exchange_rate_tutorial.yaml`:
```
version: "0.1.0"

definitions:
  selector:
    extractor:
      field_pointer: [ ]
  requester:
    url_base: "https://api.apilayer.com"
    http_method: "GET"
    authenticator:
      type: ApiKeyAuthenticator
      header: "apikey"
      api_token: "{{ config['access_key'] }}"
    request_options_provider:
      request_parameters:
        base: "{{ config['base'] }}"
  stream_slicer:
    type: "DatetimeStreamSlicer"
    start_datetime:
      datetime: "{{ config['start_date'] }}"
      datetime_format: "%Y-%m-%d"
    end_datetime:
      datetime: "{{ now_utc() }}"
      datetime_format: "%Y-%m-%d %H:%M:%S.%f"
    step: "1d"
    datetime_format: "%Y-%m-%d"
    cursor_field: "{{ options['stream_cursor_field'] }}"
  retriever:
    record_selector:
      $ref: "*ref(definitions.selector)"
    paginator:
      type: NoPagination
    requester:
      $ref: "*ref(definitions.requester)"
    stream_slicer:
      $ref: "*ref(definitions.stream_slicer)"
  base_stream:
    retriever:
      $ref: "*ref(definitions.retriever)"
  rates_stream:
    $ref: "*ref(definitions.base_stream)"
    $options:
      name: "rates"
      primary_key: "date"
      path: "/exchangerates_data/{{stream_slice['start_time'] or 'latest'}}"
      stream_cursor_field: "date"
streams:
  - "*ref(definitions.rates_stream)"
check:
  stream_names:
    - "rates"
```

### PR DESCRIPTION
**Pull request by juweins**
## What
The yaml reference files in the tutorial for config-based connectors provided wrong keywords. Therefore, a user could not follow the example properly.

## How
Adjusted the error-leading keywords on all occurences in this tutorial:

```yaml
spec: 
  documentationUrl: https://docs.airbyte.io/integrations/sources/exchangeratesapi
  connectionSpecification:
```

This PR also includes the related Issue #20047 , which also gets resolved. 

The PR will also resolve #18663. (Duplicate)

## 🚨 User Impact 🚨
Minor.

## Pre-merge Checklist
Expand the relevant checklist and delete the others.

<details><summary><strong>Updating a connector</strong></summary>
### Community member or Airbyter

- [ ] Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [x] Documentation updated
    - [x] Connector's `README.md`
- [x] PR name follows [PR naming conventions](https://docs.airbyte.com/contributing-to-airbyte/issues-and-pull-requests)

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing
- [ ] New Connector version released on Dockerhub and connector version bumped by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)

</details>

<details><summary><strong>Connector Generator</strong></summary>

- [ ] Issue acceptance criteria met
- [x] PR name follows [PR naming conventions](https://docs.airbyte.com/contributing-to-airbyte/issues-and-pull-requests)
- [x] Documentation which references the generator is updated as needed

</details>

## Tests

Manually worked through the tutorial to ensure correctness.

## Comment from Contributor
I am not able to rename commit 0f35da5d2cf7e48dd61ce14171eb797f65de97cc to reference #20047 instead of misleading issue number. Maybe you can fix this.

### CODE REVIEW

File: docs/connector-development/config-based/tutorial/5-incremental-reads.md [hunk 1]
```diff
@@ -11,8 +11,8 @@ First we'll update the spec block in `source_exchange_rates_tutorial/exchange_ra
 
 ```yaml
 spec: 
-  documentationUrl: https://docs.airbyte.io/integrations/sources/exchangeratesapi
-  connectionSpecification:
+  documentation_url: https://docs.airbyte.io/integrations/sources/exchangeratesapi
+  connection_specification:
```
Comment (marcosmarxm) [inline docs/connector-development/config-based/tutorial/5-incremental-reads.md:15]: Why are you changing this? It doesn't look correct.

**Review by marcosmarxm — CHANGES_REQUESTED** [review]

File: docs/connector-development/config-based/tutorial/5-incremental-reads.md (hunk 1 above)
Comment (juweins) [reply docs/connector-development/config-based/tutorial/5-incremental-reads.md:15]: I came accross this error by validating issue #20047 . Strictly following the tutorial example, both keywords in the yaml snippets seem to be incorrect. Changing them to the proposed naming resolves the error.

Comment (juweins) [reply docs/connector-development/config-based/tutorial/5-incremental-reads.md:15]: This is issue has been stated in #18663

Comment (marcosmarxm) [reply docs/connector-development/config-based/tutorial/5-incremental-reads.md:15]: Sorry I missed one update from low-code CDK https://github.com/airbytehq/airbyte/pull/18411/files#r1003903362

**Review by marcosmarxm — APPROVED** [review]
@brianjlai can you make a quick review of this change?

**Review by brianjlai — APPROVED** [review]
changes look good to me! thanks for the contribution 😁

**Pull request merged**